src/renderer/components/ScreenSharePicker.tsx

```diff
@@ -347,29 +417,70 @@ function ModalComponent({
-                            const height = Number(settings.resolution);
-                            const width = Math.round(height * (16 / 9));
-                            Object.assign(conn.videoStreamParameters[0], {
-                                maxFrameRate: Number(settings.fps),
-                                maxPixelCount: width * height,
-                                maxBitrate: 8000000,
-                                maxResolution: {
-                                    type: "fixed",
-                                    width,
-                                    height
-                                }
-                            });
+                        try {
+                            var conn = [...MediaEngineStore.getMediaEngine().connections].find(
+                                connection => connection.streamUserId === UserStore.getCurrentUser().id
+                            );
+                        } catch {
+                            console.log("No current stream.");
+                        }
+                        console.log([...MediaEngineStore.getMediaEngine().connections]);
+                        try {
+                            if (conn) {
+                                console.log(conn);
+                                const track = conn.input.stream.getVideoTracks()[0];
+                                console.log(track);
+                                const frameRate = Number(settings.fps);
+                                const height = Number(settings.resolution);
+                                const width = Math.round(height * (16 / 9));
+                                var constraints = track.getConstraints();
+                                const newConstraints = {
+                                    ...constraints,
+                                    frameRate,
+                                    advanced: [{ width: width, height: height }],
+                                    resizeMode: "none"
+                                };
+                                track.applyConstraints(newConstraints).then(() => {
+                                    console.log("Applied constraints from ScreenSharePicker successfully.");
+                                    console.log("New constraints:", track.getConstraints());
+                                });
+                            }
+                        } catch {
+                            console.log("No current stream.");
+                        }
+                        try {
+                            submit({
+                                    id: selected!,
+                                    ...settings
+                                });
+
+                            //reapply contraints after some time to let discord resubmit stream
+                            //i believe there MUST be way to do it cleaner..
+                            setTimeout(() => {
```

Comment:
`submit` is more than likely async.
```suggestion
                            submit({
                              id: selected!,
                              ...settings
                            }).then(() => {
```

Reply:
I retested with .then() and it doesn't work. Seems like discord doing screenres+30fps stuff after then() call. So I think that we should use timeout :c